gyp info it worked if it ends with ok
gyp info using node-gyp@10.0.1
gyp info using node@20.12.2 | darwin | x64
gyp info find Python using Python version 3.12.1 found at "/usr/local/bin/python3"
gyp info spawn /usr/local/bin/python3
gyp info spawn args [
gyp info spawn args '/usr/local/lib/node_modules/npm/node_modules/node-gyp/gyp/gyp_main.py',
gyp info spawn args 'binding.gyp',
gyp info spawn args '-f',
gyp info spawn args 'make',
gyp info spawn args '-I',
gyp info spawn args '/Users/osamatotally/Desktop/skill/backend/node_modules/mongoo/node_modules/kerberos/build/config.gypi',
gyp info spawn args '-I',
gyp info spawn args '/usr/local/lib/node_modules/npm/node_modules/node-gyp/addon.gypi',
gyp info spawn args '-I',
gyp info spawn args '/Users/osamatotally/Library/Caches/node-gyp/20.12.2/include/node/common.gypi',
gyp info spawn args '-Dlibrary=shared_library',
gyp info spawn args '-Dvisibility=default',
gyp info spawn args '-Dnode_root_dir=/Users/osamatotally/Library/Caches/node-gyp/20.12.2',
gyp info spawn args '-Dnode_gyp_dir=/usr/local/lib/node_modules/npm/node_modules/node-gyp',
gyp info spawn args '-Dnode_lib_file=/Users/osamatotally/Library/Caches/node-gyp/20.12.2/<(target_arch)/node.lib',
gyp info spawn args '-Dmodule_root_dir=/Users/osamatotally/Desktop/skill/backend/node_modules/mongoo/node_modules/kerberos',
gyp info spawn args '-Dnode_engine=v8',
gyp info spawn args '--depth=.',
gyp info spawn args '--no-parallel',
gyp info spawn args '--generator-output',
gyp info spawn args 'build',
gyp info spawn args '-Goutput_dir=.'
gyp info spawn args ]
gyp info spawn make
gyp info spawn args [ 'BUILDTYPE=Release', '-C', 'build' ]
In file included from ../lib/kerberos.cc:1:
../lib/kerberos.h:32:48: error: unknown type name 'Arguments'
   32 |   static Handle<Value> AuthGSSClientInit(const Arguments &args);
      |                                                ^
../lib/kerberos.h:33:48: error: unknown type name 'Arguments'
   33 |   static Handle<Value> AuthGSSClientStep(const Arguments &args);
      |                                                ^
../lib/kerberos.h:34:50: error: unknown type name 'Arguments'
   34 |   static Handle<Value> AuthGSSClientUnwrap(const Arguments &args);
      |                                                  ^
../lib/kerberos.h:35:48: error: unknown type name 'Arguments'
   35 |   static Handle<Value> AuthGSSClientWrap(const Arguments &args);
      |                                                ^
../lib/kerberos.h:36:49: error: unknown type name 'Arguments'
   36 |   static Handle<Value> AuthGSSClientClean(const Arguments &args);
      |                                                 ^
../lib/kerberos.h:39:34: error: unknown type name 'Arguments'
   39 |   static Handle<Value> New(const Arguments &args);  
      |                                  ^
../lib/kerberos.h:42:23: error: unknown type name 'uv_work_t'
   42 |   static void Process(uv_work_t* work_req);
      |                       ^
../lib/kerberos.h:44:21: error: unknown type name 'uv_work_t'
   44 |   static void After(uv_work_t* work_req);
      |                     ^
In file included from ../lib/kerberos.cc:3:
../lib/worker.h:17:5: error: unknown type name 'uv_work_t'
   17 |     uv_work_t request;
      |     ^
In file included from ../lib/kerberos.cc:4:
../lib/kerberos_context.h:44:34: error: unknown type name 'Arguments'
   44 |   static Handle<Value> New(const Arguments &args);  
      |                                  ^
../lib/kerberos_context.h:46:69: error: unknown type name 'AccessorInfo'
   46 |   static Handle<Value> ResponseGetter(Local<String> property, const AccessorInfo& info);  
      |                                                                     ^
../lib/kerberos_context.h:27:39: error: too few arguments to function call, single argument 'context' was not specified
   27 |     Local<Object> obj = val->ToObject();
      |                         ~~~~~~~~~~~~~ ^
/Users/osamatotally/Library/Caches/node-gyp/20.12.2/include/node/v8-value.h:382:44: note: 'ToObject' declared here
  382 |   V8_WARN_UNUSED_RESULT MaybeLocal<Object> ToObject(
      |                                            ^
  383 |       Local<Context> context) const;
      |       ~~~~~~~~~~~~~~~~~~~~~~
In file included from ../lib/kerberos.cc:4:
../lib/kerberos_context.h:27:19: error: no viable conversion from 'MaybeLocal<Object>' to 'Local<Object>'
   27 |     Local<Object> obj = val->ToObject();
      |                   ^     ~~~~~~~~~~~~~~~
/Users/osamatotally/Library/Caches/node-gyp/20.12.2/include/node/v8-local-handle.h:190:7: note: candidate constructor (the implicit copy constructor) not viable: no known conversion from 'MaybeLocal<Object>' to 'const Local<Object> &' for 1st argument
  190 | class Local {
      |       ^~~~~
/Users/osamatotally/Library/Caches/node-gyp/20.12.2/include/node/v8-local-handle.h:190:7: note: candidate constructor (the implicit move constructor) not viable: no known conversion from 'MaybeLocal<Object>' to 'Local<Object> &&' for 1st argument
  190 | class Local {
      |       ^~~~~
/Users/osamatotally/Library/Caches/node-gyp/20.12.2/include/node/v8-local-handle.h:195:13: note: candidate template ignored: could not match 'Local' against 'MaybeLocal'
  195 |   V8_INLINE Local(Local<S> that) : val_(reinterpret_cast<T*>(*that)) {
      |             ^
/Users/osamatotally/Library/Caches/node-gyp/20.12.2/include/node/v8-local-handle.h:331:22: note: explicit constructor is not a candidate
  331 |   explicit V8_INLINE Local(T* that) : val_(that) {}
      |                      ^
In file included from ../lib/kerberos.cc:4:
../lib/kerberos_context.h:28:32: error: member reference type 'Persistent<FunctionTemplate>' is not a pointer; did you mean to use '.'?
   28 |     return constructor_template->HasInstance(obj);
      |            ~~~~~~~~~~~~~~~~~~~~^~
      |                                .
../lib/kerberos_context.h:28:34: error: no member named 'HasInstance' in 'v8::Persistent<v8::FunctionTemplate>'
   28 |     return constructor_template->HasInstance(obj);
      |            ~~~~~~~~~~~~~~~~~~~~  ^
../lib/kerberos.cc:40:15: error: calling a protected constructor of class 'v8::HandleScope'
   40 |   HandleScope scope;
      |               ^
/Users/osamatotally/Library/Caches/node-gyp/20.12.2/include/node/v8-local-handle.h:104:13: note: declared protected here
  104 |   V8_INLINE HandleScope() = default;
      |             ^
../lib/kerberos.cc:41:50: error: no member named 'New' in 'v8::String'
   41 |   return ThrowException(Exception::Error(String::New(msg)));
      |                                          ~~~~~~~~^
../lib/kerberos.cc:49:15: error: calling a protected constructor of class 'v8::HandleScope'
   49 |   HandleScope scope;
      |               ^
/Users/osamatotally/Library/Caches/node-gyp/20.12.2/include/node/v8-local-handle.h:104:13: note: declared protected here
  104 |   V8_INLINE HandleScope() = default;
      |             ^
../lib/kerberos.cc:52:26: error: no matching function for call to 'New'
   52 |   constructor_template = Persistent<FunctionTemplate>::New(t);
      |                          ^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
/Users/osamatotally/Library/Caches/node-gyp/20.12.2/include/node/v8-persistent-handle.h:228:23: note: candidate function not viable: requires 2 arguments, but 1 was provided
  228 |   V8_INLINE static T* New(Isolate* isolate, Local<T> that);
      |                       ^   ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
/Users/osamatotally/Library/Caches/node-gyp/20.12.2/include/node/v8-persistent-handle.h:229:23: note: candidate function not viable: requires 2 arguments, but 1 was provided
  229 |   V8_INLINE static T* New(Isolate* isolate, T* that);
      |                       ^   ~~~~~~~~~~~~~~~~~~~~~~~~~
fatal error: too many errors emitted, stopping now [-ferror-limit=]
20 errors generated.
make: *** [Release/obj.target/kerberos/lib/kerberos.o] Error 1
gyp ERR! build error 
gyp ERR! stack Error: `make` failed with exit code: 2
gyp ERR! stack at ChildProcess.<anonymous> (/usr/local/lib/node_modules/npm/node_modules/node-gyp/lib/build.js:209:23)
gyp ERR! System Darwin 24.0.0
gyp ERR! command "/usr/local/bin/node" "/usr/local/lib/node_modules/npm/node_modules/node-gyp/bin/node-gyp.js" "rebuild"
gyp ERR! cwd /Users/osamatotally/Desktop/skill/backend/node_modules/mongoo/node_modules/kerberos
gyp ERR! node -v v20.12.2
gyp ERR! node-gyp -v v10.0.1
gyp ERR! not ok 
